microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -155,79 +134,54 @@ func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient
 		if res == "" || !strings.Contains(res, "kind:") {
 			continue
 		}
-		// if create failed, wait 2s and retry, this will be removed when the monitor task is implemented
-		for {
-			createdObj, err := applyResourceToK8s(ctx, dynamicClient, ns, []byte(res))
-			if err != nil {
-				fmt.Printf("Failed to reconcile resource: %v\n", err)
-			} else {
-				fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
+		createdObj, err := applyResourceToK8s(ctx, client, ns, svc, []byte(res))
+		if err != nil {
+			return fmt.Errorf("Failed to reconcile resource: %v\n", err)
+		} else {
+			fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
 
-				// return the service obj to get the serivce URL from it
-				if retSvc != nil && createdObj.GetKind() == "Service" {
-					err = scheme.Scheme.Convert(createdObj, retSvc, nil)
-					if err != nil {
-						fmt.Printf("Failed to save service: %v\n", err)
-					}
+			// return the service obj to get the service URL from it
+			if retSvc != nil && createdObj.GetKind() == Service {
+				err = scheme.Scheme.Convert(createdObj, retSvc, nil)
+				if err != nil {
+					return fmt.Errorf("Failed to save service: %v\n", err)
 				}
+			}
 
-				if createdObj.GetKind() == "Deployment" {
-					var newEnvVars []corev1.EnvVar
-					if svcCfg != nil {
-						for name, value := range *svcCfg {
-							if name == "endpoint" {
-								continue
-							}
-							itemEnvVar := corev1.EnvVar{
-								Name:  name,
-								Value: value,
-							}
-							newEnvVars = append(newEnvVars, itemEnvVar)
-						}
-					}
-					if len(newEnvVars) > 0 {
-						deployment := &appsv1.Deployment{}
-						err = runtime.DefaultUnstructuredConverter.FromUnstructured(createdObj.UnstructuredContent(), deployment)
-						if err != nil {
-							fmt.Printf("Failed to save deployment: %v\n", err)
+			if createdObj.GetKind() == Deployment && step != Router {
```

Comment:
Why do we skip the `Router` step? I believe that Router also need to set the environment variable, am I right?

Reply:
I think for router we can set the env by patching the "gmc-router.yaml" in patchCustomConfigToTemplates, no need to do it here.